src/modules/file/file.usecase.ts

```diff
@@ -39,4 +54,25 @@ export class FileService {
       deletedAt: new Date(),
     });
   }
+
+  async getEncryptionKeyFileFromShare(
+    fileId: string,
+    network: any,
+    share: Share,
+    code: string,
+  ) {
+    const encryptedMnemonic = share.mnemonic.toString();
+    const mnemonic = aes.decrypt(encryptedMnemonic, code);
+    const { index } = await network.getFileInfo(share.bucket, fileId);
+    const encryptionKey = await Environment.utils.generateFileKey(
+      mnemonic,
+      share.bucket,
+      Buffer.from(index, 'hex'),
+    );
+    return encryptionKey.toString('hex');
+  }
+
+  async getTotalSizeOfFilesFromFolder(folderId: number) {
+    return await this.fileRepository.getTotalSizeByFolderId(folderId);
```

Comment:
`async-await` could be avoided if you return a `promise`, which makes code more readable.